aea/crypto/base.py

```diff
@@ -165,3 +166,14 @@ def is_transaction_settled(self, tx_digest: str) -> bool:
         :param tx_digest: the digest associated to the transaction.
         :return: True if the transaction has been settled, False o/w.
         """
+
+    @abstractmethod
+    def validate_transaction(self, tx_digest: str, proposal_msg: Message) -> bool:
```

Comment:
Why are we passing the entire message? I think we should pass the values we need so we don't create unnecessary coupling between these modules.